runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp

```diff
@@ -23,13 +23,94 @@
 #include "codegen/ARM64PrivateLinkage.hpp"
 #include "codegen/Linkage_inlines.hpp"
 
-TR::ARM64LinkageProperties TR::ARM64PrivateLinkage::properties =
+TR::ARM64PrivateLinkage::ARM64PrivateLinkage(TR::CodeGenerator *cg)
+   : TR::Linkage(cg)
    {
-   };
+   int32_t i;
+
+   _properties._properties = 0;
+
+   _properties._registerFlags[TR::RealRegister::NoReg] = 0;
+   _properties._registerFlags[TR::RealRegister::x0]    = IntegerArgument|IntegerReturn;
+   _properties._registerFlags[TR::RealRegister::x1]    = IntegerArgument;
+   _properties._registerFlags[TR::RealRegister::x2]    = IntegerArgument;
+   _properties._registerFlags[TR::RealRegister::x3]    = IntegerArgument;
+   _properties._registerFlags[TR::RealRegister::x4]    = IntegerArgument;
+   _properties._registerFlags[TR::RealRegister::x5]    = IntegerArgument;
+   _properties._registerFlags[TR::RealRegister::x6]    = IntegerArgument;
+   _properties._registerFlags[TR::RealRegister::x7]    = IntegerArgument;
+
+   for (i = TR::RealRegister::x8; i <= TR::RealRegister::x15; i++)
+      _properties._registerFlags[i] = 0; // x8 - x15 volatile
+
+   _properties._registerFlags[TR::RealRegister::x16]   = ARM64_Reserved; // IP0
+   _properties._registerFlags[TR::RealRegister::x17]   = ARM64_Reserved; // IP1
+
+   for (i = TR::RealRegister::x18; i <= TR::RealRegister::x28; i++)
+      _properties._registerFlags[i] = Preserved; // x18 - x28 Preserved
+
+   _properties._registerFlags[TR::RealRegister::x29]   = ARM64_Reserved; // FP
+   _properties._registerFlags[TR::RealRegister::x30]   = ARM64_Reserved; // LR
+   _properties._registerFlags[TR::RealRegister::sp]    = ARM64_Reserved;
+   _properties._registerFlags[TR::RealRegister::xzr]   = ARM64_Reserved;
```

Comment:
* x19 (vmthread) and x20 (Java stack pointer) should also be reserved

* in fact, both registers should be "locked" (sorry, I missed that before I merged #5567).  Would you mind fixing that.

Reply:
Reserved and locked `x19` and `x20`.